sdk/samples/iot/provisioning/src/paho_iot_provisioning_example.c

```diff
@@ -321,135 +330,168 @@ static int register_device()
+  {
+    printf("SUCCESS - Device provisioned:\n");
+    printf("\tHub Hostname: ");
+    print_az_span(response.registration_result.assigned_hub_hostname);
+    printf("\tDevice Id: ");
+    print_az_span(response.registration_result.device_id);
+  }
+  else // Unsuccesful assignment (unassigned, failed or disabled states)
+  {
+    printf("ERROR - Device Provisioning failed:\n");
+    printf("\tRegistration state: ");
+    print_az_span(response.operation_status);
+    printf("\tLast operation status: %d\n", response.status);
+    printf("\tOperation ID: ");
+    print_az_span(response.operation_id);
+    printf("\tError code: %u\n", response.registration_result.extended_error_code);
+    printf("\tError message: ");
+    print_az_span(response.registration_result.error_message);
+    printf("\tError timestamp: ");
+    print_az_span(response.registration_result.error_timestamp);
+    printf("\tError tracking ID: ");
+    print_az_span(response.registration_result.error_tracking_id);
+
+    if (response.retry_after_seconds > 0)
     {
-      // Successful assignment - print out the assigned hostname and device id
-      if (operation_status == AZ_IOT_PROVISIONING_STATUS_ASSIGNED)
-      {
-        printf("SUCCESS - Device provisioned:\n");
-        printf("\tHub Hostname: ");
-        print_az_span(response.registration_result.assigned_hub_hostname);
-        printf("\tDevice Id: ");
-        print_az_span(response.registration_result.device_id);
-      }
-      else // unassigned, failed or disabled states
-      {
-        printf("ERROR - Device Provisioning failed:\n");
-        printf("\tRegistration state: ");
-        print_az_span(response.operation_status);
-        printf("\tLast operation status: %d\n", response.status);
-        printf("\tOperation ID: ");
-        print_az_span(response.operation_id);
-        printf("\tError code: %u\n", response.registration_result.extended_error_code);
-        printf("\tError message: ");
-        print_az_span(response.registration_result.error_message);
-        printf("\tError timestamp: ");
-        print_az_span(response.registration_result.error_timestamp);
-        printf("\tError tracking ID: ");
-        print_az_span(response.registration_result.error_tracking_id);
-        if (response.retry_after_seconds > 0)
-        {
-          printf("\tRetry-after: %u seconds.", response.retry_after_seconds);
-        }
-      }
+      printf("\tRetry-after: %u seconds.", response.retry_after_seconds);
     }
+  }
 
-    MQTTClient_freeMessage(&message);
-    MQTTClient_free(topic);
+  return get_operation_status_free(MQTTCLIENT_SUCCESS, topic, message);
+}
+
+static az_result parse_operation_message(
+    char* topic,
+    int topic_len,
+    MQTTClient_message* message,
+    az_iot_provisioning_client_register_response* response,
+    az_iot_provisioning_client_operation_status* operation_status)
+{
+  int rc;
+
+  printf("Topic: %s\n", topic);
+
```

Comment:
what happens length is non-zero?

Reply:
I'm not sure here.  I will ask Cris about it.  Good question.